robot_driver/src/inverse_dynamics.cpp

```diff
@@ -107,6 +112,25 @@ bool InverseDynamicsController::computeLegCommandArray(
                                     ref_foot_acceleration, grf_array,
                                     contact_mode, tau_array);
 
+    // Convert gains to eigen vectors for easier math
+    Eigen::VectorXd swing_kp_cart_eig, swing_kd_cart_eig,
+        swing_cart_fb(3 * num_feet_);
+    quad_utils::vectorToEigen(swing_kp_cart_, swing_kp_cart_eig);
+    quad_utils::vectorToEigen(swing_kd_cart_, swing_kd_cart_eig);
+
+    // Compute PD feedback in cartesian space
+    Eigen::MatrixXd jacobian(3 * num_feet_, state_positions.size());
+    swing_cart_fb = swing_kp_cart_eig.replicate<4, 1>().cwiseProduct(
+                        ref_foot_positions - foot_positions) +
+                    swing_kd_cart_eig.replicate<4, 1>().cwiseProduct(
+                        ref_foot_velocities - foot_velocities);
+
+    // Transform PD into joint space
+    quadKD_->getJacobianBodyAngVel(state_positions, jacobian);
+    swing_cart_fb =
+        jacobian.block(0, 0, 3 * num_feet_, 3 * num_feet_).transpose() *
+        swing_cart_fb;
```

Comment:
I don't think this is the correct jacobian for mapping the cartesian space force to joint torque. The jacobian is 12*18 that maps [body state, joint state] to [foot state].